torchtune/training/_distributed.py

```diff
@@ -608,16 +679,25 @@ def shard_model(
             the forward pass. Setting this to True corresponds to the FULL_SHARD sharding strategy
             from FSDP1, while setting it to False corresponds to the SHARD_GRAD_OP sharding strategy.
 
+    Raises:
+        ValueError: If no layer modules were sharded. Please check if shard conditions are working as expected.
```

Comment:
nit: the 2nd sentence is just a copy-paste from the error message? Would just remove it